app/HutchManager/Data/Entities/DataSource.cs

```diff
@@ -1,9 +1,12 @@
+using System.Text.Json.Serialization;
+
 namespace HutchManager.Data.Entities;
 
 public class DataSource
 {
   public string Id { get; set; } = string.Empty;
   public DateTimeOffset LastCheckin { get; set; }
 
+  [JsonIgnore]
```

Comment:
I'm guessing that because this is in `Data/Entities` it is the database table. What you might want to do instead of `[JsonIgnore]` here, is to have a `Dto` class for this entity which doesn't have the `Agents` field. Then in the controller, create an instance of that `Dto` to be returned as a JSON (which will then not have the agents list).

Reply:
100% this. Don't return Entities from Controllers, or otherwise serialise them to JSON tbh. Instead use a new type that represents the appropriate model for the interaction you're working on (e.g. returning a list of summaries of items).

The only thing I'd change from what @dcl10 said is put it in `Models/` not `Dto/`.

while it could be called a DTO (Data Transfer Object), the convention we seem to have arrived at in this codebase is that models for interactions with the Manager's API are in `Models/` and models for interactions with external APIs (e.g. RQuest) are in `Dto/`